sys/stdio/stdio.c

```diff
@@ -42,7 +42,14 @@ void stdio_init(void)
 ssize_t stdio_write(const void* buffer, size_t len)
 {
     for (unsigned i = 0; i < XFA_LEN(stdio_provider_t, stdio_provider_xfa); ++i) {
-        stdio_provider_xfa[i].write(buffer, len);
+        size_t cursor = 0;
+        while (cursor < len) {
+            ssize_t written = stdio_provider_xfa[i].write(&buffer[written], len);
+            if (written < 0) {
+                break;
+            }
+            cursor += written;
+        }
```

Comment:
Didn't we say that we *don't* want to do this because with some back-ends, this would block forever if stdio is written from ISR?

Reply:
In an ISR situation, a backend will (or should) only do one of two things: return an error, or return the full length. Either terminates the loop.